phases/19-capstone-projects/83-prompt-injection-detector/code/main.py

```
@@ -0,0 +1,247 @@
+from typing import Iterable
+
+from benign import prompts as load_benign
+from rules import REGEX_RULES, SUBSTRING_RULES, all_rules
+
+HERE = Path(__file__).parent
+OUTPUTS = HERE.parent / "outputs"
+TAXONOMY_PATH = HERE.parent.parent / "82-jailbreak-taxonomy" / "outputs" / "taxonomy.json"
+
+LEET_TABLE = str.maketrans({"0": "o", "1": "i", "3": "e", "4": "a", "5": "s", "7": "t", "@": "a", "$": "s"})
+ZERO_WIDTH = re.compile(r"[​‌‍⁠‪-‮]")
+HOMOGLYPHS = str.maketrans({"А": "A", "В": "B", "С": "C", "Е": "E", "Н": "H", "К": "K", "М": "M", "О": "O", "Р": "P", "Т": "T", "Х": "X"})
```

Ruff (0.15.14)

[error] Contains control characters that can permit obfuscated code

(PLE2502)